components/centraldashboard/app/api.ts

```diff
@@ -70,18 +69,42 @@ export class Api {
             })
         .get(
           '/dashboard-links',
-          async (_: Request, res: Response) => {
+          async (req: Request, res: Response) => {
+            const acceptlanguage = req.header('accept-language');
             const linksData = await this.k8sService.getDashboardLinks();
-            let links = {};
+            let langLinks = {};
             try {
-              links=JSON.parse(linksData.data["links"]);
+              const links = JSON.parse(linksData.data["links"]);
+              const lang = this.resolveLanguage
+              (
+                this.getBrowserLanguages(acceptlanguage), 
+                Object.getOwnPropertyNames(links), 
+                linksData.data["defaultLanguage"]
+              );
+              langLinks = links[lang];
             }catch(e){
               return apiError({
                 res, code: 500,
                 error: ERRORS.invalid_links_config,
               });
             }
-            res.json(links);
+            res.json(langLinks);
           });
   }
+
+  resolveLanguage(requested: string[], supported: string[], defaultLang: string) {
+    return requested.find(lang => supported.includes(lang)) || defaultLang;
+  }
+
+  getBrowserLanguages(lang: string) {
+    const languages = lang.split(',');
+    if (!languages) {
+      return [];
+    }
+    // Append fallbacks not explicit in browser languages.
+    // Non-destructive: string keys will be reported back in order of insertion.
+    let languagelist: {[code: string]: boolean};
+    languagelist = Object.fromEntries([].concat(languages).map(lang => [lang.split(/-|_|;/)[0], true]));
```

Comment:
Good call on adding `;`. MDN doesn't show underscore as being a valid character for this header. Do some browsers use it in place of hyphen?

Reply:
No, accidentally left that in there. Thanks!